server/controllers/file.controller.js

```diff
@@ -7,64 +7,82 @@ import { deleteObjectsFromS3, getObjectKey } from './aws.controller';
 // Bug -> timestamps don't get created, but it seems like this will
 // be fixed in mongoose soon
 // https://github.com/Automattic/mongoose/issues/4049
-export function createFile(req, res) {
-  Project.findOneAndUpdate(
-    {
-      _id: req.params.project_id,
-      user: req.user._id
-    },
-    {
-      $push: {
-        files: req.body
-      }
-    },
-    {
-      new: true
-    }, (err, updatedProject) => {
-      if (err || !updatedProject) {
-        console.log(err);
-        res.status(403).send({ success: false, message: 'Project does not exist, or user does not match owner.' });
-        return;
-      }
-      const newFile = updatedProject.files[updatedProject.files.length - 1];
-      updatedProject.files.id(req.body.parentId).children.push(newFile.id);
-      updatedProject.save((innerErr) => {
-        if (innerErr) {
-          console.log(innerErr);
-          res.json({ success: false });
-          return;
+export async function createFile(req, res) {
+  try {
+    let updatedProject = await Project.findOneAndUpdate(
+      {
+        _id: req.params.project_id,
+        user: req.user._id
+      },
+      {
+        $push: {
+          files: req.body
         }
-        res.json(updatedProject.files[updatedProject.files.length - 1]);
+      },
+      {
+        new: true
+      }
+    );
+
+    if (!updatedProject) {
+      res.status(403).send({
+        success: false,
+        message: 'Project does not exist, or user does not match owner.'
       });
+      return;
     }
-  );
+    const newFile = updatedProject.files[updatedProject.files.length - 1];
+    updatedProject.files.id(req.body.parentId).children.push(newFile.id);
+    try {
```

Comment:
especially nested try/catch... i would try to avoid it if possible.

Reply:
Is there any way to avoid them?

Reply:
@catarak @siddhant1 I have a suggestion that can clear up this try catch litter. We can define a wrapper function to catch error and wrapper every async function with it. This way we can avoid try/catch with async await.


```js
/* Helper buddy for removing async/await try/catch litter.  
 We can make the name of this function a little smaller */
function catchError(promise) {
  return promise.then(data => {
    if (data instanceof Error) return [data]
    return [null, data]
  }).catch(err => [err])
}

// Example Usage
async function usageExample(params) {
  const [ err, data ] = await catchError(myPromise(params)) // We are using the wrapper here

  if (err) {
    // handle error
  }
  // Do stuff with data
}
```


What do you think about it?

Reply:
hmmmm, i prefer `express-async-handler` as i like that it handles the error handling as middleware. to me, it feels like a cleaner solution, rather than adding a helper function to every single controller function.

Reply:
Express-async-handler is good 
I will refactor it along with adding a better error handling approach , because for now we are just logging the errors

Reply:
i recommend not just pushing ahead on something. i think there are a few different options here  and i'd like to see a comparison of these options, so we can make a decision about what the best path forward is.